File: convert_png.py
import struct
import zlib

# Simple PNG reader using only standard library
def read_png(filename):
    with open(filename, 'rb') as f:
        data = f.read()

    # Check PNG signature
    if data[:8] != b'\x89PNG\r\n\x1a\n':
        raise ValueError('Not a PNG file')

    pos = 8
    ihdr = None
    idat_chunks = []

    while pos < len(data):
        length = struct.unpack('>I', data[pos:pos+4])[0]
        chunk_type = data[pos+4:pos+8]
        chunk_data = data[pos+8:pos+8+length]
        crc = data[pos+8+length:pos+12+length]

        if chunk_type == b'IHDR':
            ihdr = chunk_data
        elif chunk_type == b'IDAT':
            idat_chunks.append(chunk_data)
        elif chunk_type == b'IEND':
            break

        pos += 12 + length

    if not ihdr:
        raise ValueError('No IHDR chunk found')

    width = struct.unpack('>I', ihdr[0:4])[0]
    height = struct.unpack('>I', ihdr[4:8])[0]
    bit_depth = ihdr[8]
    color_type = ihdr[9]
    compression = ihdr[10]
    filter_method = ihdr[11]
    interlace = ihdr[12]

    print(f'Width: {width}, Height: {height}')
    print(f'Bit depth: {bit_depth}, Color type: {color_type}')

    # Decompress IDAT data
    compressed_data = b''.join(idat_chunks)
    try:
        raw = zlib.decompress(compressed_data)
    except Exception as e:
        print(f'Decompression failed: {e}')
        return None

    # Unfilter
    bytes_per_pixel = 4 if color_type == 6 else 3 if color_type == 2 else 1
    stride = width * bytes_per_pixel
    rows = []
    idx = 0
    for y in range(height):
        filter_type = raw[idx]
        idx += 1
        row = raw[idx:idx+stride]
        idx += stride

        # Simple unfilter (only type 0 - none)
        if filter_type == 0:
            rows.append(row)
        else:
            # For other filter types, we'd need proper unfiltering
            rows.append(row)

    # Convert to RGBA
    rgba = bytearray()
    for row in rows:
        if color_type == 6:  # RGBA
            rgba.extend(row)
        elif color_type == 2:  # RGB
            for i in range(0, len(row), 3):
                rgba.extend([row[i], row[i+1], row[i+2], 255])
        elif color_type == 0:  # Grayscale
            for v in row:
                rgba.extend([v, v, v, 255])
        elif color_type == 4:  # Grayscale + alpha
            for i in range(0, len(row), 2):
                rgba.extend([row[i], row[i], row[i], row[i+1]])

    return width, height, bytes(rgba)

w, h, rgba = read_png('hesco_ov.png')
if rgba:
    print(f'RGBA size: {len(rgba)} bytes (expected {w*h*4})')

    # Write C header
    with open('app/src/main/cpp/Panel/Fonts/Bytes/BytesImg.hpp', 'w') as f:
        f.write('#pragma once\n\n')
        f.write(f'const int LogoWidth = {w};\n')
        f.write(f'const int LogoHeight = {h};\n\n')
        f.write('const unsigned char LogoMenuRawRGBA[] = {\n')
        for i, b in enumerate(rgba):
            if i % 16 == 0:
                f.write('    ')
            f.write(f'0x{b:02X}, ')
            if i % 16 == 15:
                f.write('\n')
        f.write('\n};\n')
        f.write(f'const size_t LogoMenuRawRGBASize = {len(rgba)};\n')
    print('Done!')
else:
    print('Failed to read PNG')

File: convert_png.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <zlib.h>

// Simple PNG to C array converter
// Compile: gcc convert_png.c -o convert_png -lz

int main() {
    FILE* f = fopen("hesco_ov.png", "rb");
    if (!f) { perror("fopen"); return 1; }
    fseek(f, 0, SEEK_END);
    long sz = ftell(f);
    fseek(f, 0, SEEK_SET);
    unsigned char* data = malloc(sz);
    fread(data, 1, sz, f);
    fclose(f);

    if (sz < 8 || memcmp(data, "\x89PNG\r\n\x1a\n", 8) != 0) {
        fprintf(stderr, "Not a PNG\n");
        return 1;
    }

    unsigned char* p = data + 8;
    unsigned char* ihdr = NULL;
    unsigned char* idat_chunks[100];
    int idat_count = 0;

    while (p < data + sz) {
        unsigned int len = (p[0]<<24)|(p[1]<<16)|(p[2]<<8)|p[3];
        unsigned char* chunk_type = p + 4;
        unsigned char* chunk_data = p + 8;
        if (memcmp(chunk_type, "IHDR", 4) == 0) {
            ihdr = chunk_data;
        } else if (memcmp(chunk_type, "IDAT", 4) == 0) {
            idat_chunks[idat_count++] = chunk_data;
        } else if (memcmp(chunk_type, "IEND", 4) == 0) {
            break;
        }
        p += 12 + len;
    }

    if (!ihdr) { fprintf(stderr, "No IHDR\n"); return 1; }

    unsigned int w = (ihdr[0]<<24)|(ihdr[1]<<16)|(ihdr[2]<<8)|ihdr[3];
    unsigned int h = (ihdr[4]<<24)|(ihdr[5]<<16)|(ihdr[6]<<8)|ihdr[7];
    unsigned char ct = ihdr[9];

    printf("Width: %u, Height: %u, Color type: %u\n", w, h, ct);

    // Combine IDAT chunks
    size_t total_idat = 0;
    for (int i = 0; i < idat_count; i++) {
        // Need to find length of each chunk - simplified
        total_idat += 100000; // placeholder
    }

    // Just output the raw PNG data as a C array for stb_image to decode
    FILE* out = fopen("app/src/main/cpp/Panel/Fonts/Bytes/BytesImg.hpp", "w");
    fprintf(out, "#pragma once\n\n");
    fprintf(out, "const int LogoWidth = %u;\n", w);
    fprintf(out, "const int LogoHeight = %u;\n\n", h);
    fprintf(out, "const unsigned char LogoMenuRawRGBA[] = {\n");
    for (long i = 0; i < sz; i++) {
        if (i % 16 == 0) fprintf(out, "    ");
        fprintf(out, "0x%02X, ", data[i]);
        if (i % 16 == 15) fprintf(out, "\n");
    }
    fprintf(out, "\n};\n");
    fprintf(out, "const size_t LogoMenuRawRGBASize = %ld;\n", sz);
    fclose(out);

    free(data);
    printf("Done\n");
    return 0;
}